### ISSUE TITLE
[JOSS Review] Paper questions and editorial nitpicks

### ISSUE THREAD

**Issue by matthewfeickert** [Infineon/StreamGen#7]
This issue is with regards to the JOSS review happening at https://github.com/openjournals/joss-reviews/issues/7206.

Overall, I want to say that you did a great job of capturing the spirit of what a JOSS paper is supposed to be, so thank you for that! Along these lines though, do you feel that Figure 2

https://github.com/Infineon/StreamGen/blob/30d7f2deeede8612e9e7a6ea996eacc6ca510cb6/docs/paper.md?plain=1#L52

is strictly necessary? Or is Figure 1 sufficient and then the two places where Figure 2 is referred to 

https://github.com/Infineon/StreamGen/blob/30d7f2deeede8612e9e7a6ea996eacc6ca510cb6/docs/paper.md?plain=1#L23

https://github.com/Infineon/StreamGen/blob/30d7f2deeede8612e9e7a6ea996eacc6ca510cb6/docs/paper.md?plain=1#L33

could just refer to the documentation? If you feel Figure 2 is necessary / critically relevant that's totally fine, but it might be useful to reduce the figure caption in that case to a more succinct summary.

Now getting into the truly nitpicky bits, there are a few places where a hyphen is used where an em-dash should be used instead.

* `represent the *domain* - the data distribution common to all classes.` in

https://github.com/Infineon/StreamGen/blob/30d7f2deeede8612e9e7a6ea996eacc6ca510cb6/docs/paper.md?plain=1#L52C243-L52C312

* `from past experiences - a phenomenon referred to as` in

https://github.com/Infineon/StreamGen/blob/30d7f2deeede8612e9e7a6ea996eacc6ca510cb6/docs/paper.md?plain=1#L57C1-L58C1

In these situations the hyphen (-) should be replaced with an em-dash (&mdash;). As the papers are being powered by Markdown and not pure LaTeX we can't reach for the easy LaTeX `---` but you can use `&mdash;` which will insert an em-dash (note the `;` at the end which is actually relevant).

* As https://github.com/c0fec0de/anytree doesn't have a `CITATION.cff` there's no easy citation there for you to take (that's fine) but for

https://github.com/Infineon/StreamGen/blob/30d7f2deeede8612e9e7a6ea996eacc6ca510cb6/docs/paper.md?plain=1#L38

is there a particular release version that is relevant to the mention in the paper that could be cited? Or just the entire project?

* Can you try to escape with quotes 

https://github.com/Infineon/StreamGen/blob/30d7f2deeede8612e9e7a6ea996eacc6ca510cb6/docs/paper.bib#L154

so that the authors show up as "TorchVision maintainers and contributors" in the references instead of "maintainers, T., & contributors"?

* I see in the review that `editorialbot` is giving a skipped DOI for

```
- No DOI given, and none found for title: PyTorch: An Imperative Style, High-Performance Dee...
```

given that 

https://github.com/Infineon/StreamGen/blob/30d7f2deeede8612e9e7a6ea996eacc6ca510cb6/docs/paper.bib#L144-L150

is in the `paper.bib` but apparently not used. If PyTorch doesn't need to be directly cited can you remove that from the `paper.bib` to avoid these catches?

Other than that I think this is great. :+1: Feel free to ask questions if you have any!

**Comment by matthewfeickert** [comment]
By the way, if it is at all useful to preview changes to the paper locally rather than having `editorialbot` rebuild it (which to be clear, is fine) you can use Docker to build the paper on your local machine using these instructions: https://joss.readthedocs.io/en/latest/paper.html#docker

**Comment by LaurenzBeck** [comment]
> Overall, I want to say that you did a great job of capturing the spirit of what a JOSS paper is supposed to be

thank you @matthewfeickert 🙏, this means a lot to me.

I will go through your comments now.

**Comment by LaurenzBeck** [comment]
> In these situations the hyphen (-) should be replaced with an em-dash (—).

Nice catch, I just learned about this last week in a scientific writing seminar. I will change the hyphens 💪

**Comment by LaurenzBeck** [comment]
> is in the paper.bib but apparently not used. If PyTorch doesn't need to be directly cited can you remove that from the paper.bib to avoid these catches?

removed 😊

**Comment by LaurenzBeck** [comment]
> Can you try to escape with quotes so that the authors show up as "TorchVision maintainers and contributors" in the references instead of "maintainers, T., & contributors"?

Quotes resulted in "maintainers, TorhVision, and contributors" (or similar).
Surrounding the field with double { (author = {{{TorchVision maintainers and contributors}}},) did the trick 😊

**Comment by LaurenzBeck** [comment]
I have a formatting question regarding the list of affiliations at the beginning of the paper:

Is there a way to include a linebreak between the two entries?
I would prefer something like:

Laurenz A. Farthofer 1,2
1) KAI - Kompetenzzentrum Automobil- und Industrieelektronik GmbH, Austria 
2) Institute of Computer Graphics and Vision, Graz University of Technology, Austria

**Comment by LaurenzBeck** [comment]
Ok, I checked anytree and I do not depend on the major changes introduced in the current 2.x versions.
Therefore, I would like to cite the project as a whole.

How do I need to adapt the citation entry?

**Comment by LaurenzBeck** [comment]
I had a look at the caption of Figure 2:
I don't need any additional space, so I personally don't mind the longer caption. (I always prefer self-contained figures)
If you would still prefer a shorter one, we could remove the detailed explanation after the first sentence in the caption.

I tried and it looks very acceptable with the statement of need section starting on the second page. I could try 1 or two tricks/rephrasings to save ~3  lines, which would result in the last page only containing references. (Now it also contains the last parahraph from the acknowledgements...
What do you think @matthewfeickert ?

**Comment by matthewfeickert** [comment]
> Is there a way to include a linebreak between the two entries?

Good question. I'm not sure, but I'll ask and if support for this gets added post publication we have the ability to rerender a paper.

> Ok, I checked anytree and I do not depend on the major changes introduced in the current 2.x versions. Therefore, I would like to cite the project as a whole.

In your `pyproject.toml` you require the latest release to date of the project

https://github.com/Infineon/StreamGen/blob/30d7f2deeede8612e9e7a6ea996eacc6ca510cb6/pyproject.toml#L32

You can select whatever version requirements you'd like, but if you decided to be more library-like then this would ideally be an empirically tested lower bound. `anytree` doesn't make citation super easy given that they don't provide any citation guidance or long term archives with DOIs, but from [_Software citation principles_](https://peerj.com/articles/cs-86/) (2016) one of the motivations of software citation is specificity

> ... Software identification should be as specific as necessary, such as using version numbers, revision numbers, or variants such as platforms.

So I'd be inclined to cite what you require, which regardless of if you're actually using the features, in StreamGen's case is `v2.12.1`.

> I don't need any additional space, so I personally don't mind the longer caption. (I always prefer self-contained figures). If you would still prefer a shorter one, we could remove the detailed explanation after the first sentence in the caption.

I'd suggest keeping it as short as possible to explain **only** the _need for the software_. I am in agreement with you that figures should give the necessary context themselves, but at seven sentences long this is a full paragraph on technical information that the documentation should cover in full. If the documentation doesn't then I'd suggest moving the detailed explanation here to the documentation and then simply giving a 1-2 sentence summary and citing the docs.

**Comment by LaurenzBeck** [comment]
good, I will check the lowest version for which my test suit passes and then update both the dependency in `pyproject.toml`, as well as the reference in `paper.bib` :)

I will also shorten the figure caption and check if the docs contain the detailed explanations.
Should I explicitly refer to the docs for more details in the figure caption, or is this obvious?

Thank you for raising the question in inara 💪

**Comment by LaurenzBeck** [comment]
ok, I checked and unfortunately, I depend on anytree being >=2.7 since I make use of the `UniqueDotExporter`, which was introduced then. I lowered the dependency constrain to >=2.7.0

I also added package versions to anytree and torchvision, which look great in the references.

I also shortened the figure caption and removed some whitespace/new-lines in the "future work" section to have the last page only show the references.

Finally, I did a patch release of streamgen to make the lower dependency constraints applicable for end users :)

How long do we wait for an answer from inara#95 @matthewfeickert ?

**Comment by matthewfeickert** [comment]
> ok, I checked and unfortunately, I depend on anytree being >=2.7 since I make use of the `UniqueDotExporter`, which was introduced then. I lowered the dependency constrain to >=2.7.0
> 
> I also added package versions to anytree and torchvision, which look great in the references.
> 
> I also shortened the figure caption and removed some whitespace/new-lines in the "future work" section to have the last page only show the references.
> 
> Finally, I did a patch release of streamgen to make the lower dependency constraints applicable for end users :)

Great. I'll rerender on https://github.com/openjournals/joss-reviews/issues/7206 to check it out.

> How long do we wait for an answer from inara#95

I wouldn't delay publication on this. If it isn't quickly addressed I'll make sure the rerender happens post publication if there is a fix.

**Comment by matthewfeickert** [comment]
Looks like everything has been resolved as of 9d63df38faac7072ada0fa4e81d60c85c542d46d, so closing. :+1:
